packages/core/core/src/PackagerRunner.js

```diff
@@ -119,8 +119,31 @@ export default class PackagerRunner {
     |} = {};
     let writeEarlyPromises = {};
     let hashRefToNameHash = new Map();
-    // skip inline bundles, they will be processed via the parent bundle
-    let bundles = bundleGraph.getBundles().filter(bundle => !bundle.isInline);
+    let bundles = bundleGraph.getBundles().filter(bundle => {
+      // Do not package and write placeholder bundles to disk. We just
+      // need to update the name so other bundles can reference it.
+      if (bundle.isPlaceholder) {
+        invariant(
+          !this.options.shouldContentHash,
+          'Lazy bundling does not work with content hashing',
+        );
```

Comment:
Should this be asserted up front in `resolveOptions`?
So `invariant(!(options.shouldBuildLazily && options.shouldContentHash));`